modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/helper/DefaultCommerceCheckoutStepHttpHelper.java

```diff
@@ -594,6 +594,9 @@ private CommerceOrder _updateCommerceOrder(
 			commerceShippingOptionKey, commerceContext,
 			_portal.getLocale(httpServletRequest));
 
+		commerceOrder = _commerceOrderLocalService.recalculatePrice(
```

Comment:
This should be inside a transaction with _commerceOrderLocalService.updateCommerceShippingMethod....
See ShippingMethodCommerceCheckoutStep.java at line 283